src/eth/types.ts

```diff
@@ -11,3 +15,87 @@ export const EthUint256Struct = definePattern(
   'EthUint256',
   /^0x([1-9a-f][0-9a-f]*|0)$/iu,
 );
+
+/**
+ * Supported Ethereum methods.
+ */
+export enum EthMethod {
+  // General signing methods
+  PersonalSign = 'personal_sign',
+  Sign = 'eth_sign',
+  SignTransaction = 'eth_signTransaction',
+  SignTypedDataV1 = 'eth_signTypedData_v1',
+  SignTypedDataV3 = 'eth_signTypedData_v3',
+  SignTypedDataV4 = 'eth_signTypedData_v4',
+}
+
+/**
+ * Supported Ethereum methods for ERC-4337 (Account Abstraction) accounts.
+ */
+export enum EthErc4337Method {
+  // ERC-4337 methods
+  PrepareUserOperation = 'eth_prepareUserOperation',
+  PatchUserOperation = 'eth_patchUserOperation',
+  SignUserOperation = 'eth_signUserOperation',
+}
+
+/**
+ * Supported Ethereum account types.
+ */
+export enum EthAccountType {
+  Eoa = 'eip155:eoa',
+  Erc4337 = 'eip155:erc4337',
+}
```

Comment:
Similarly to another comments, should we always have acronyms capitalized unless we're using snake case?

```suggestion
export enum EthAccountType {
  EOA = 'eip155:eoa',
  ERC4337 = 'eip155:erc4337',
}
```